jre/javatests/com/google/j2cl/jre/java8/util/stream/CollectorsTest.java

```diff
@@ -591,6 +591,76 @@ public void testSet() {
     assertSingleItemCollectedAs(Collections.singleton("a"), c, "a");
   }
 
+  public void testToUnmodifiableList() {
+    applyItems(List.of("a", "b"), toUnmodifiableList(), "a", "b");
+    assertUnmodifiableCollection(applyItemsWithSplitting(toUnmodifiableList(), "a", "b"), "a", "z");
+    try {
+      Stream.of("a").map(ignore -> null).collect(toUnmodifiableList());
+      fail("Expected NPE");
+    } catch (NullPointerException ignore) {
+      // expected
+    }
+  }
+
+  public void testToUnmodifiableMap() {
+    // verify simple cases copy all values and results are unmodifiable
+    applyItems(Map.of("a", 0, "b", 1), toUnmodifiableMap(Function.identity(),
+        k -> k.charAt(0) - 'a'), "a", "b");
+    Map<String, Integer> tested = applyItemsWithSplitting(toUnmodifiableMap(Function.identity(),
+        k -> k.charAt(0) - 'a'), "a", "b");
+    assertUnmodifiableMap(tested, "a", 0, "z", 100);
+
+    // verify merge works with only one key (but this is just passing through to the toMap func
+    // anyway...)
+    applyItems(Map.of("a", 2), toUnmodifiableMap(Function.identity(), ignore -> 1, Integer::sum),
+        "a", "a");
+
+    // verify nulls blow up for both keys and values
+    try {
+      Stream.of("a").collect(toUnmodifiableMap(obj -> null, Function.identity()));
+      fail("Expected NPE");
+    } catch (NullPointerException ignore) {
+      // expected
+    }
+    try {
+      Stream.of("a").collect(toUnmodifiableMap(Function.identity(), obj -> null));
+      fail("Expected NPE");
+    } catch (Exception ignore) {
+      // expected
+    }
+  }
+
+  private <K, V> void assertUnmodifiableMap(Map<K, V> a, K existingKey, V existingValue, K newKey,
```

Comment:
Let's move this helper next to assertUnmodifiableCollection.

Reply:
done